module-2/backend/week_two.md

```diff
@@ -26,29 +34,81 @@ end
+      6. The model response to the controller with necessary attributes/methods
+      7. The controller then response to the user and shows a desired view they requested, with the correct info from the 
+      model.
+11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques. 
+      simpliying the way the methods in the model are calling on the datatbase 
+      in your controller, setting all the instances of a thing then in the views calling all of the methods on that instance
+      setting a layout view so that all views follow a basic layout then sending all the views forms to there so that there is 
+      less repitition
+      
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
-2. Name your three favorite ActiveRecord rake tasks and describe what they do.
-4. What can you expect from a group as you begin working together? As you continue working together?
-5. What two columns does `t.timestamps null: false` create in our database?
-6. What cURL flag can you use to send a `POST` request?
-7. What case does JSON (and JavaScript) use for multi-word variables?
+      #pluck : returns the attribute you pass in 
+      #where : finds the object where the passed in conditions are true 
+      #find_or_create_by : finds an object and if it isn't there it returns it 
+      #create  : makes a new instance AND saves it 
+      #find_by : finds object by whatever you pass in. 
+2. Name your three favorite ActiveRecord rake tasks and describe what they do. 
+      rake db:seed - populates the database 
+      rake db:drop - you can go back and delete migrations and change them 
+      rake db:rollback  - goes back one migration ago so you can go back and change or delete the last migration
+4. What can you expect from a group as you begin working together? As you continue working together? 
+      These stages: Forming, Storming, Norming, Performing, Adjerning 
+      and where you are in one stage doesn't mean youll always either stay or move forward. You can also move back 
+      to storming for example. 
+5. What two columns does `t.timestamps null: false` create in our database? 
+      created at column and updated at column 
+6. What cURL flag can you use to send a `POST` request? 
+        curl -X POST
+7. What case does JSON (and JavaScript) use for multi-word variables? 
+        lower_case_snack_case
 8. What case does Ruby use for multi-word variables?
+        same
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+        schools ---- teachers
+        one ------------many
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+        schools       teachers
+        --------      --------
+        teacher_id    
```

Comment:
Careful here! Assuming a teacher belongs to a school (and a school has many teachers) the school_id would go on teacher.